### PR TITLE
Add Hover link and node feature

### PR DESCRIPTION
**Pull request by goliath-walker**
Feat: On link hover, highlight source and target note incl. labels #58 
Feat: On note hover, for all links highlight source and target note incl. labels

Minor Fix: Properly toggle graph mode (https://github.com/treymo/joplin-link-graph/compare/main...goliath-walker:feat-hover?expand=1#diff-0ab86666b99a06c57c509a8b80148b035d7273ea88847a2776138529f3435d6fL57-R69)
Minor Feat: Optimized default style for distance-based graph mode
Refactor: Distinguish more clearly between graph mode styles in CSS file

### CODE REVIEW

File: src/webview.css [hunk 1]
```diff
@@ -41,30 +39,33 @@ circle {
...
-  stroke: var(--distance-1-primary-color);
+.node-label {
+  fill: var(--joplin-color);
 }
 
-.adjacent-line {
-  stroke: var(--distance-0-primary-color);
-  stroke-width: 3px;
+.hovered-node {
+  fill: var(--joplin-color-error) !important;
```
Comment (treymo) [inline src/webview.css:59]: I love the changes, but my only comment is that the red error color seems aggressive for the graph.  At least in the Light and Dark themes I tried it in.

What are your thoughts on using something like `joplin-color-correct` or `joplin-color-warn`?    i.e. green or yellow

Comment (treymo) [reply src/webview.css:59]: after testing locally, I think I prefer green (`joplin-color-correct`).

Comment (goliath-walker) [reply src/webview.css:59]: Yes, that is a great idea.
Currently, there are two colors used: one for the nodes with more contrast and one for rest (label, links, arrows).

I found something like #93f500 to be good for fomer, and `joplin-color-correct` with latter. So let's change it?

Comment (goliath-walker) [reply src/webview.css:59]: Ideally I would like to let users them selves change styles by creating an additional menu entry for custom CSS - that could be PR in future

Comment (treymo) [reply src/webview.css:59]: Let’s go with those new colors you suggested.

Yeah an interface for customizing colors in future would be awesome!

Comment (goliath-walker) [reply src/webview.css]: Nice, thanks for the quick answer! Allright, changes done.
When having time, I'll probably give a short intro of new features in the discourse forum (or you, if you want)

**Review by treymo — APPROVED** [review]

File: src/webview.css (hunk 1 above)
Comment (treymo) [reply src/webview.css]: Looks good, merged!   I just kicked off a release, definitely feel free to shout out the new features. I did something similar a couple days ago but you can do the same

**Pull request merged**